### ISSUE TITLE
Does not properly collapse sequential code blocks when opening bracket of second is on same line as closing of first

### ISSUE THREAD

**Issue by christopherjbaker** [digitallyserviced/semanticolor#34]
This has been bugging me for a while, so I decided to open an issue.

When viewing code where a sequential block starts on the same line as the previous ends, such as an if-else (see first image), collapsing the first block also removes the ability to collapse subsequent blocks (see second image). When using the built-in javascript language this is not an issue (see third image).
![image](https://user-images.githubusercontent.com/980338/63830333-c0113380-c920-11e9-9f45-82b058876537.png)
![image](https://user-images.githubusercontent.com/980338/63830380-df0fc580-c920-11e9-94b8-ed7b223f28f5.png)
![image](https://user-images.githubusercontent.com/980338/63830427-f6e74980-c920-11e9-9d26-55c20af1ccf7.png)

The built in language only supports this if tree-sitters are enabled, so I assume it is being held up by your http://atom/atom#19151, but I wanted to point out another issue with the lack of tree-sitter support.

**Comment by christopherjbaker** [comment]
I understand why the other issue has been closed (though sadly) as it is not language agnostic. I do not understand why this is being closed, however. This is an objective bug in how these languages compare to others.

**Comment by sharedprophet** [comment]
As you said, it will be fixed when tree-sitter support is possible. Hopefully that will be soon? I have created a branch of atom which would enable it and a branch on here that is ready to merge if they do pull in my suggestion. Until then, there's not much I can do about this.

**Comment by christopherjbaker** [comment]
oh ok. Without comment, it looked like it was "closed because not gonna happen", but its more "closed because we can't until this other thing". That's fair. I did start following that other issue too. Is there anything I can do over there to encourage them to support this?

**Comment by sharedprophet** [comment]
Not that I know of. I just proposed this change earlier today, so let's give them a chance to look at it before we bug them too much. : )

**Comment by christopherjbaker** [comment]
Is that a 100% solution? I saw that some of your previous ones were only close.

**Comment by sharedprophet** [comment]
Yes, I got it working really well with that solution, and it should avoid the performance hit Max Brunsfeld was concerned about.
